### PR TITLE
Add PM4 support

### PR DESCRIPTION
**Pull request by thebigsmileXD**

### CODE REVIEW

**Comment by thebigsmileXD** [comment]
So apparently there is no option on GitHub to request creation of a new branch when opening a pull request :/

This PR would remove PM3 support entirely. If you can, create a new branch called PM4 on GitHub and then i will re-PR to that branch

**Comment by andresbytes** [comment]
Good job! I created the PM4 branch.

File: src/NpcDialog/PacketListener.php [hunk 1]
```diff
@@ -23,14 +23,14 @@ class PacketListener implements Listener {
 
     public function onPacketReceiveEvent(DataPacketReceiveEvent $event): void {
         $packet = $event->getPacket();
-        $player = $event->getPlayer();
-        $server = $player->getServer();
+		$server = Server::getInstance();
```
Comment (andresbytes) [inline src/NpcDialog/PacketListener.php:26]: The indentation of these lines is wrong.

File: src/NpcDialog/PacketListener.php [hunk 2]
```diff
@@ -23,14 +23,14 @@ class PacketListener implements Listener {
...
+		$player = $event->getOrigin()->getPlayer();
 
-        if(!($packet instanceof NpcRequestPacket) or ($entity = $server->findEntity($packet->entityRuntimeId)) === null) {
+        if($player === null || !($packet instanceof NpcRequestPacket) || ($entity = $server->getWorldManager()->findEntity($packet->entityRuntimeId)) === null) {
             return;
         }
 
-        $username = $player->getName();
+		$username = $player->getName();
```
Comment (andresbytes) [inline src/NpcDialog/PacketListener.php:33]: Wrong indentation here too.

**Review by andresbytes — CHANGES_REQUESTED** [review]

File: src/NpcDialog/PacketListener.php (hunk 1 above)
Comment (thebigsmileXD) [reply src/NpcDialog/PacketListener.php:26]: Wonder how that happened, let's blame PhpStorm. It looks like tabs and spaces mixed. I will patch up manually

**Pull request closed without merging**